dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hive/AbstractHiveConnector.java

```diff
@@ -64,21 +63,28 @@ public abstract class AbstractHiveConnector extends AbstractConnector {
     @ThreadSafe
     public static class ThriftClientPool implements AutoCloseable {
 
+        public interface ThriftClientConsumer {
+            void accept(HiveMetastoreThriftClient thriftClient) throws Exception;
+        }
+
         @Nonnull
         private final String name;
         @Nonnull
         private final ThreadLocal<? extends HiveMetastoreThriftClient> threadLocalThriftClient;
         @Nonnull
+        private final ExecutorManager executorManager;
+        @Nonnull
         private final ExecutorService executorService;
         @Nonnull
         private final Object lock = new Object();
         @GuardedBy("lock")
         @Nonnull
         private final List<@NonNull HiveMetastoreThriftClient> builtClients = new ArrayList<>();
 
-        public ThriftClientPool(@Nonnull String name, @Nonnull HiveMetastoreThriftClient.Builder thriftClientBuilder, @Nonnull ExecutorService executorService) {
+        public ThriftClientPool(@Nonnull String name, @Nonnull HiveMetastoreThriftClient.Builder thriftClientBuilder, int threadPoolSize) {
             this.name = Preconditions.checkNotNull(name, "name was null.");
-            this.executorService = Preconditions.checkNotNull(executorService, "executorService was null.");
+            this.executorService = ExecutorManager.newExecutorServiceWithBackpressure(name, threadPoolSize);
```

Comment:
Interesting. I might have still passed in the ExecutorService, as things like caller-thread-only/blocking ExecutorService(s) exist, and are useful for testing. Actually, this isn't critical, as newESWB returns a DirectExecutor if threadPoolSize is 1.